### DISCUSSION TITLE
Sort channels by logical channel number where available, alphabetically where missing

### DISCUSSION THREAD

**Discussion by loucasal** [iptv-org/community#511] [Ideas]
Another QoL/user friendliness improvement already adopted by other repos, in the same vein of iptv-org/iptv#6133.
Rather than having all channels sorted alphabetically, we could sort at least those that match DVB channels according to their logical channel number (LCN) by default. Those that do not have one could keep being sorted alphabetically.

Some examples:
https://en.wikipedia.org/wiki/List_of_DTT_channels_in_the_United_Kingdom
https://en.wikipedia.org/wiki/Television_in_Italy#List_of_nationwide_television_stations

**Comment by Dum4G** [comment]
LCN is kinda variable thing that depends on region and your ISP. Pluto TV may for example have it's own set of LCN that will conflict to something common for another source and a general playlist.

**Reply by loucasal** [reply]
Yes, of course it wouldn't make sense for all channels. However where channels broadcast nationwide are also available via IPTV, many users would find it natural to have them sorted following their LCN.

**Reply by what-a** [reply]
For over the air channels in the US, LCNs are unique only within a local market. 
Once you gain a national view and see streams from multiple local markets, the LCNs no longer provide value and can lead to confusion (e.g. channel 4.1 from one local market has nothing to do with 4.1 from another).

**Comment by GreenSnakeLinux** [comment]
> Another QoL/user friendliness improvement already adopted by other repos, in the same vein of iptv-org/iptv#6133. Rather than having all channels sorted alphabetically, we could sort at least those that match DVB channels according to their logical channel number (LCN) by default. Those that do not have one could keep being sorted alphabetically.
> 
> Some examples: https://en.wikipedia.org/wiki/List_of_DTT_channels_in_the_United_Kingdom https://en.wikipedia.org/wiki/Television_in_Italy#List_of_nationwide_television_stations
@loucasal Do you have more links like this for others countries ?
As a workaround you can add numbers before the name of channels to sort them, then remove them in your xml parsing app.

**Reply by loucasal** [reply]
> @loucasal Do you have more links like this for others countries ?

Wikipedia has a handful of them: at a glance, I see FR, IT, PL, UK.
This page is also a good starting point: https://en.wikipedia.org/wiki/Virtual_channel
